### PR TITLE
[Test] Should be able to use free shipping voucher in checkout

### PR DESCRIPTION
**Pull request by szczecha**
I want to merge this change because it adds a test for a free shipping voucher in checkout [CORE_0903](https://saleor.testmo.net/repositories/5?group_id=131&case_id=1400)

# Impact

- [ ] New migrations
- [ ] New/Updated API fields or mutations
- [ ] Deprecated API fields or mutations
- [ ] Removed API types, fields, or mutations

# Docs

- [ ] Link to documentation:

# Pull Request Checklist

- [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
- [ ] Database queries are optimized and the number of queries is constant
- [ ] Database migrations are either absent or optimized for zero downtime
- [ ] The changes are covered by test cases

### CODE REVIEW

**Review by yellowee — CHANGES_REQUESTED** [review]

File: saleor/tests/e2e/checkout/test_checkout_with_free_shipping_voucher.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+    warehouse_id, channel_id, channel_slug, shipping_method_id = prepare_shop(
+        e2e_staff_api_client
+    )
+
+    _, product_variant_id, product_variant_price = prepare_product(
+        e2e_staff_api_client, warehouse_id, channel_id, variant_price=25
+    )
+
+    _voucher_discount_value, voucher_code = prepare_free_shipping_voucher(
```
[the remark this answers is not in the document]
Comment (szczecha) [reply saleor/tests/e2e/checkout/test_checkout_with_free_shipping_voucher.py:74]: Good catch I will fix it.

File: saleor/tests/e2e/checkout/test_checkout_with_free_shipping_voucher.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    checkout_lines = checkout_data["lines"][0]
+    shipping_method_id = checkout_data["shippingMethods"][0]["id"]
+
+    assert checkout_data["isShippingRequired"] is True
+    assert checkout_lines["unitPrice"]["gross"]["amount"] == float(
+        product_variant_price
+    )
+    assert checkout_lines["undiscountedUnitPrice"]["amount"] == product_variant_price
+
+    # Step 2 Add voucher code to checkout
```
[the remark this answers is not in the document]
Comment (szczecha) [reply saleor/tests/e2e/checkout/test_checkout_with_free_shipping_voucher.py:106]: In this case, the voucher has some limitations: minimal spent amount = 75USD. So as a user, I want to use the voucher for free shipping but when I enter the code it results in an error due to too low total of the chart. So I increased the quantity of products, added shipping, and then tried to add code one again and it resulted in free shipping.
I think it's a common case and can be included in e2e.

[the remark this answers is not in the document]
Comment (yellowee) [reply saleor/tests/e2e/checkout/test_checkout_with_free_shipping_voucher.py:106]: Ok, I was not aware of the voucher's limitations - thank you for the explanation!

[the remark this answers is not in the document]
Comment (szczecha) [reply saleor/tests/e2e/checkout/test_checkout_with_free_shipping_voucher.py:106]: I will think about better test naming :D

**Review by fowczarek — APPROVED** [review]

**Review by yellowee — APPROVED** [review]

**Pull request merged**